clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -71,12 +71,12 @@
     /**
      * @see KafkaConsumer#commit(Map, CommitType)
      */
-    public void commit(Map<TopicPartition, Long> offsets, CommitType commitType);
+    public void commit(Map<TopicPartition, OffsetMetadata> offsets, CommitType commitType);
```

Comment:
> On the other hand, maybe most users don't even specify the offsets manually anyway and the concern here is unwarranted since 99% of the cases are handled by commit(CommitType) and commit(CommitType, ConsumerCommitCallback)

Agree with @ewencp that manual offset commit is really a very small percentage of all uses. Even though I agree that amongst that minority, fewer would have custom metadata, I'm not sure it is worth adding the extra commitWithMetadata API for. 

It may be ok in this case to go with 
```public void commit(Map<TopicPartition, OffsetMetadata> offsets, CommitType commitType); ```